### PR TITLE
Include a systemd service with our rpm

### PR DESCRIPTION
**Pull request by jbowes**
The service can be configured with machine token credentials in
/etc/torus/token.environment, for use in production environments.

Other processes running on the system can access secrets from torus by:
- running as a user in the `torus` group
- setting the TORUS_ROOT environment variable to /var/run/torus

While there is a torus user, this user should *not* be used for running
processes that need access to secrets. It is an isolated sandbox that
has access to the machine's token credentials.

### CODE REVIEW

File: daemon/socket/proxy.go [hunk 1]
```diff
@@ -48,10 +48,10 @@ type AuthProxy struct {
 
 // NewAuthProxy returns a new AuthProxy. It will return an error if creation
 // of the domain socket fails, or the upstream registry URL is misconfigured.
-func NewAuthProxy(c *config.Config, sess session.Session, db *db.DB,
```
Comment (ianlivingstone) [inline daemon/socket/proxy.go]: Can we get a doc string about the different the socket is configured (read/write user/group)?

File: contrib/systemd/torus.service [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+Description=Torus daemon
+Documentation=https://www.torus.sh/docs/
+After=network.target
+
+[Service]
+Type=simple
+User=torus
+Environment="TORUS_ROOT=/var/run/torus"
+EnvironmentFile=/etc/torus/token.environment
+ExecStart=/usr/bin/torus daemon start --foreground --no-permission-check
```
Comment (ianlivingstone) [inline contrib/systemd/torus.service]: So in "service mode" anyone on the system gets access to any of the secrets that the systems identity provides it?

Comment (ianlivingstone) [inline contrib/systemd/torus.service]: I see -- it's constrained to the group. Does this also mean that they can access the TOKEN_ID and TOKEN_SECRET via `/proc` for the running user? Or is the idea that the user is locked down, then you just share secrets via group?

File: daemon/socket/proxy.go (hunk 1 above)
Comment (jbowes) [inline daemon/socket/proxy.go]: *nods*

File: contrib/systemd/torus.service (hunk 2 above)
Comment (ianlivingstone) [inline contrib/systemd/torus.service]: 👍 Awesome.

**Pull request merged**